src/Morphir/JsonSchema/Backend.elm

```diff
@@ -105,55 +105,84 @@ extractTypes modName definition =
+                                    ( typeName, Const (ctorName |> Name.toTitleCase) )
+
+                                else
+                                    ( typeName
+                                    , Array
+                                        (TupleType
+                                            (Const (ctorName |> Name.toTitleCase)
+                                                :: (ctorArgs
+                                                        |> List.map (Tuple.second >> mapType)
+                                                   )
+                                            )
+                                        )
+                                    )
+                            )
+                   )
+
+
+mapType : Type ta -> SchemaType
 mapType typ =
     case typ of
-        Type.Reference a fQName types ->
-            case FQName.toString fQName of
-                "Morphir.SDK:Basics:int" ->
-                    Just Integer
+        Type.Reference _ fQName argTypes ->
+            case ( FQName.toString fQName, argTypes ) of
+                ( "Morphir.SDK:Basics:int", [] ) ->
+                    Integer
 
-                "Morphir.SDK:String:string" ->
-                    Just String
+                ( "Morphir.SDK:String:string", [] ) ->
+                    String
 
-                "Morphir.SDK:Basics:float" ->
-                    Just Number
+                ( "Morphir.SDK:Basics:float", [] ) ->
+                    Number
 
-                "Morphir.SDK:Basics:bool" ->
-                    Just Boolean
+                ( "Morphir.SDK:Basics:bool", [] ) ->
+                    Boolean
 
-                "Morphir.SDK:List:list" ->
-                    List.head types
-                        |> Maybe.andThen mapType
-                        |> Maybe.map Array
+                ( "Morphir.SDK:List:list", [ itemType ] ) ->
+                    Array (ListType (mapType itemType))
 
                 _ ->
-                    Nothing
+                    Ref ("#/defs/" ++ (fQName |> FQName.getLocalName |> Name.toTitleCase))
 
-        Type.Record a fields ->
+        Type.Record _ fields ->
             fields
                 |> List.foldl
-                    (\field maybeDictSoFar ->
-                        maybeDictSoFar
-                            |> Maybe.map2
-                                (\schemaType dictSofar ->
-                                    Dict.insert (Name.toTitleCase field.name) schemaType dictSofar
-                                )
-                                (mapType field.tpe)
+                    (\field ->
+                        \dictSofar ->
+                            Dict.insert (Name.toTitleCase field.name) (mapType field.tpe) dictSofar
                     )
-                    (Just Dict.empty)
-                |> Maybe.map Object
+                    Dict.empty
+                |> Object
 
         _ ->
-            Nothing
+            Null
```

Comment:
This does not look right. What this mapping is saying is that for any types that we don't support the expected type in the target JSON document is null. The right way of handling this would be to report an error.

Reply:
Fixed now. Added support for null

Reply:
This doesn't solve the problem I raised. The problem is that you are mapping to `null` instead of reporting an error.